Batteries/Data/Vector/Basic.lean

```diff
@@ -0,0 +1,249 @@
+import Batteries.Data.Array
+import Batteries.Data.List.Basic
+import Batteries.Data.List.Lemmas
+
+namespace Vector
+
+open Array
+
+/--
+  `Vector α n` is an `Array α` whose size is statically fixed to `n`
+-/
+def Vector (α : Type u) (n : Nat) := {a : Array α // a.size = n}
+
+/--
+  `Vector.size` gives the size of a vector. It is the vector counterpart to
+  `Array.size`
```

Comment:
Not sure how useful the last sentence is, when it's rather obvious (i.e. same name). Also, a user of `Vector` may not care too much about `Array` itself.

Reply:
@nomeata I fixed that docstring. It doesn't refer to Array.size anymore